### ISSUE TITLE
[Bug] - Can not find class from logs in terminal

### ISSUE THREAD

**Issue by mamilic** [redhat-developer/vscode-java#3822]
When I try to navigate to classname by CTRL+Left Click from the logs it can not find it.

##### Environment
- Operating System: Windows 11
- JDK version: 21
- Visual Studio Code version: 1.94.2
- Java extension version: 1.35.1

##### Steps To Reproduce
1. Open [demo.zip](https://github.com/user-attachments/files/17411531/demo.zip)
2. Run the DemoApplication.java
3. Try to CTRL + Left Click on com.example.demo.DemoApplication from the logs
![image](https://github.com/user-attachments/assets/65adf22b-0b8f-4704-b95b-7ab85f6a937d)
4. It will open input box with that full classname, however as it is full path it can not find it, only DemoApplication can be found.
![image](https://github.com/user-attachments/assets/a4187a3c-0a8f-47cf-89bd-a88a9a00c532)

**Comment by rgrunber** [comment]
Yup, I see the same behaviour. The problem is in how the value shows up in the command palette. It's being inserted directly without the `#` at the beginning. That `#` is used to indicate it's a workspace symbol search. You can confirm this by doing `Go -> Go To Symbol In Workspace` and you'll see the command palette comes up with a `#` .

Initially I thought there's not much we can do since it seems like something VS Code handles. However, I looked inside vscode-java-debug (I assume you have this installed), I found : https://github.com/microsoft/vscode-java-debug/blob/main/src/terminalLinkProvider.ts#L18-L60 .

It looks like the current implementation only creates links for stacktrace elements. @jdneo does it make sense to move this into vscode-java-debug and add this as a proposed enhancement there ? Is the current link coming from some default detection in VS Code itself ? Detecting fully qualified names might be tricky though.
